internal/controller/dynamic_resource_request_controller.go

```diff
@@ -156,31 +155,31 @@ func (r *DynamicResourceRequestController) Reconcile(ctx context.Context, req ct
 		return ctrl.Result{}, nil
 	}
 
+	opts := opts{client: r.Client, ctx: ctx, logger: logger}
+
+	if !rr.GetDeletionTimestamp().IsZero() {
+		logging.Info(logger, "deleting resource request")
+		return r.deleteResources(opts, promise, rr)
+	}
```

Comment:
question: if a resource has a binding, and the promise delete pipeline has changed, this will run the latest delete pipeline instead of the versioned delete pipeline. Or am I missing something?